components/core/src/clp_s/CommandLineArguments.cpp

```diff
@@ -492,6 +496,12 @@ CommandLineArguments::parse_arguments(int argc, char const** argv) {
                 );
             }
 
+            if (m_print_ordered_chunk_stats && false == m_ordered_decompression) {
+                throw std::invalid_argument(
+                        "print-ordered-chunk-stats must be used with ordered argument"
+                );
+            }
```

Comment:
Perhaps combine all the `false == m_ordered_decompression` conditions into a single block? I.e.,

* if m_ordered_decompression:
  * if 0 != target_ordered_chunk_size -> error
  * if ordered_chunk_stats -> error
  * etc.